app/controllers/api/services_controller.rb

```diff
@@ -18,6 +18,20 @@ def edit_resource(type, id, data)
       super(type, id, attributes)
     end
 
+    def add_resource_resource(type, id, data)
+      svc = resource_search(id, type, collection_class(type))
+      data['resources'].each do |resource_ref|
+        resource_type, resource_id = parse_href(resource_ref['href'])
+        resource = resource_search(resource_id, resource_type, collection_class(resource_type))
+        raise "Cannot assign #{resource_type} to #{service_ident(svc)}" unless resource.respond_to? :add_to_service
+        resource.add_to_service(svc)
```

Comment:
normal pattern as is an action result showing success/failure for each action. we do bend the rules on bulk assignment of tags on vms via the assign_tags, in which case we return one per assignment for the same reason you mentioned here above, see app/controllers/api/subcollections/tags.rb's assign_tags method.  We could do something similar here.